krpc-code-gen/src/test/java/io/kroxylicious/krpccodegen/main/KrpcGeneratorTest.java

```diff
@@ -32,75 +44,113 @@
 class KrpcGeneratorTest {
 
     private static final String MESSAGE_SPECS_PATH = "message-specs/common/message";
-    private static final String OUTPUT_DIR = "krpc-test-sources/krpc";
     private static final String TEST_CLASSES_DIR = "test-classes";
 
     @Test
-    public void testHelloWorld() throws Exception {
+    public void testHelloWorld(@TempDir File tempDir) throws Exception {
         KrpcGenerator gen = KrpcGenerator.single()
                 .withMessageSpecDir(getMessageSpecDir())
                 .withMessageSpecFilter("*.json")
                 .withTemplateDir(getTemplateDir())
                 .withTemplateNames(List.of("hello-world/example.ftl"))
                 .withOutputPackage("com.foo")
-                .withOutputDir(getOutputDir())
+                .withOutputDir(tempDir)
                 .withOutputFilePattern("${messageSpecName}.txt")
                 .build();
 
         gen.generate();
+
+        File file = new File(tempDir, new StringJoiner(File.separator).add("com").add("foo").add("FetchRequest.txt").toString());
+        assertFileHasExpectedContents(file, "hello-world/example-expected-FetchRequest.txt");
     }
 
     @Test
-    public void testKrpcData() throws Exception {
+    public void testKrpcData(@TempDir File tempDir) throws Exception {
         KrpcGenerator gen = KrpcGenerator.single()
                 .withMessageSpecDir(getMessageSpecDir())
                 .withMessageSpecFilter("*.json")
                 .withTemplateDir(getTemplateDir())
                 .withTemplateNames(List.of("Data/example.ftl"))
                 .withOutputPackage("com.foo")
-                .withOutputDir(getOutputDir())
+                .withOutputDir(tempDir)
                 .withOutputFilePattern("${messageSpecName}.java")
                 .build();
 
         gen.generate();
+
+        File file = new File(tempDir, new StringJoiner(File.separator).add("com").add("foo").add("FetchRequest.java").toString());
+        assertFileHasExpectedContents(file, "Data/example-expected-FetchRequest.java.txt");
     }
 
     @Test
-    public void testKproxyFilter() throws Exception {
+    public void testKproxyFilter(@TempDir File tempDir) throws Exception {
         KrpcGenerator gen = KrpcGenerator.single()
                 .withMessageSpecDir(getMessageSpecDir())
                 .withMessageSpecFilter("*.json")
                 .withTemplateDir(getTemplateDir())
                 .withTemplateNames(List.of("Kproxy/Filter.ftl"))
                 .withOutputPackage("com.foo")
-                .withOutputDir(getOutputDir())
+                .withOutputDir(tempDir)
                 .withOutputFilePattern("${messageSpecName}Filter.java")
                 .build();
 
         gen.generate();
+
+        File file = new File(tempDir, new StringJoiner(File.separator).add("com").add("foo").add("FetchRequestFilter.java").toString());
```

Comment:
maybe extract a method to generate path and create the file object as its repeated a few times.

Reply:
done, refactored to do `private static File join(File dir, String... pathElements)` instead